### ISSUE TITLE
A Y.Text instance merges lines with "\n" even they have different attributes in favour of the last one

### ISSUE THREAD

**Issue by NickMitin** [yjs/yjs#275]
**Describe the bug**
A Y.Text instance merges lines with "\n" even they have different attributes in favour of the last one

**To Reproduce**

Apply following ops to a Y.Text instance using .applyDelta(delta.ops)
e.g.

```javascript
delta.ops = [
    { insert: "Text" },
    { insert: "\n", attributes: { title: true} },
    { insert: "\nText" },
    { insert: "\n", attributes: { title: true} },
]
```

Read back the Y.Text delta using .toDelta()
Result will be:
```javascript
 [
    { insert: 'Text' },
    { insert: '\n\nText', attributes: { title: true } },
    { insert: '\n', attributes: { title: true } }
  ]
```

**Expected behavior**
```javascript
[
    { insert: "Text" },
    { insert: "\n", attributes: { title: true} },
    { insert: "\nText" },
    { insert: "\n", attributes: { title: true} },
]
```
**Environment Information**

Browser: Chrome, Firefox
Yjs version: yjs@13.4.9, quill@1.3.7
Additional context
yjs/y-quill#6

Issue appeared after following commit:
https://github.com/yjs/yjs/commit/186f7140b622aeda4b4a924dc9323513b2ca13c6

Related bug:
https://github.com/yjs/yjs/issues/277

**Easy copy-paste test:**
```javascript
const testText = ydoc.getText('test');

testText.applyDelta([
    { insert: "Text" },
    { insert: "\n", attributes: { title: true} },
    { insert: "\nText" },
    { insert: "\n", attributes: { title: true} },
]);
console.log(testText.toDelta());
```

**Comment by boschDev** [comment]
#277 seems like a duplicated ticket

**Comment by NickMitin** [comment]
@dmonad, hey! we (Conspectus.io) are now sponsoring your project as:
**Special Attention** I will respond to your issues and questions more promptly.

So, can you please help us to figure out how to fix this bug?

**Comment by NickMitin** [comment]
@dmonad thanks for the fix! You saved our project from disaster. Seems to be working correct now.

**Comment by dmonad** [comment]
Hi @NickMitin, thanks for sponsoring me btw! I just published the fix in 13.5.0 (with some other big updates)
